doc/ni_by_example_2.md

```diff
@@ -0,0 +1,395 @@
+
+* `lc`
+* `uc`
+* `substr`
+* `split`
+* `join`
+* `**`: exponent
+* `my $<v> = <expr>`: instantiate a scalar `<v>` with the value of `<expr>`
+* `map`
+* `keys %h`
+* Regular Expressions
+  * `$<v> =~ /regex/`
+  * `$<v> =~ s/regex//`
+  * `$<v> = tr/regex//d`
+  * `$<v> = y/regex//`
+
+
+
+
+##Useful `ni`-specific Perl Subroutines
+The operators in this section refer specifically to the 
+`$ ni <data> p'...'`
+
+* `ghe`: geohash encoding
+  * `ghe($lat, $lng, $precision)`
+    * If `$precision > 0`, returns a geohash with `$precision` base-32 characters of precision. 
+    * If `$precision < 0`, returns a geohash with `$precision` (base-2) bits of precision.
+* `ghd`: geohash decoding
+  * `ghd($gh_base32)`
+     * Returns the corresponding latitude and longitude (in that order) of the southwesternmost point corresponding to that geohash.
+  * `ghd($gh_int, $precision)`
+    * If the number of bits of precision is specified, `ghd` will decode the input integer as a geohash with $precision bits. Returns the  latitude and longitude (in that order) of the southwesternmost point corresponding to that geohash.
+* `tpe`: time parts to epoch
+  * `tpe(@time_pieces)`: Returns the epoch time and assumes that the pieces are year, month, day, hour, minute, and second, in that order.
+  * `tpe($time_format, @time_pieces)`: Returns the epoch time, using `$time_format` to determine what the ordered `@time_pieces` are.
+* `tep`: time epoch to parts
+  * `tep($epoch_time)`: returns the year, month, day, hour, minute, and second in human-readable formatfrom the epoch time.
+  * `tep($time_format, $epoch_time)`: returns the specified parts of the date using following `$time_format`.
+* `timezone_seconds`
+  * `tep($raw_timestamp + $timezone_seconds($lat, $lng))` returns the approximate date and time at the location `$lat, $lng` at a Unix timestamp of `$raw_timestamp`.
+  
+  
+##Plotting with `ni --js`
+Check out the [tutorial](tutorial.md) for some examples of cool, interactive `ni` plotting.
+
+**TODO**: Say something useful.
+
+
+##Data Closure Basics and Array-Based Operators
+Data closures are useful in that they travel with `ni` when `ni` is sent somewhere else, for example over ssh, or as a jar to a Hadoop cluster. Importantly, closures can be accessed from within Perl snippets by using their name.
+
+* `closure_name::[...]`: Create a data closure
+  * Any legal `ni` snippet that is executable on the machine from whose context `ni` is being executed.
+  * The closure can be referenced within a Perl snippet as  `p' ... closure_name ...'`
+* `a_` through `l_`: Array-based line operations 
+  * Data closures are transferred as an array of lines; in order to access data from a specific column of the data closure, you will need to use array-based line operators `a_` through `l_`, which are the array-based analogs to the line-based operators `a/a()` through `l/l()`.
```

Comment:
Actually they're multiline strings, but `a_` will behave the same way. The distinction might be important in some cases though.